Add several functions to work with refGene exon sequences

alumi opened the pull request:
#### Summary
This PR adds several utility functions to `varity.ref-gene` module.

- refGene record => region
- refGene record => sequence of exons
- refGene record => base sequence

#### Tests

- `lein check` 🆗
- `lein test :all` 🆗

#### Notes
- Upgraded dependencies
- Replaced `varity.util/revcomp-bases` with more efficient implementation in cljam 0.6.0

totakke commented on src/varity/ref_gene.clj:
```diff
@@ -137,6 +138,73 @@
+             :end e,
+             :reverse? reverse?})))))
+
+(defn cds-exon-seq
```
I think `cds-seq` is correct name. `cds-exon-` is wordy because CDS does not include introns.

alumi commented:
@totakke Done, thanks for your review! 😸

federkasten commented on src/varity/ref_gene.clj:
```diff
@@ -137,6 +139,73 @@
+  (cond-> (cseq/read-sequence seq-rdr exon)
+    reverse? util-seq/revcomp))
+
+(defn ^String read-gene-sequence
```
The `gene` word contains more widely meanings. Thus I think you should use `transcript` here.

Additionally, could you add docstrings to clarify the function, such as `the sequence contains 5'-UTR, CDS, and 3'-UTR`.

alumi commented:
@federkasten I have replaced 'gene' with 'transcript' or 'ref-gene-record'.

Also I've added a commit to fix \:strand values (please refer to https://github.com/chrovis/cljam/pull/143#discussion_r213180699)
I'll open another PR for fixing existing functions like https://github.com/chrovis/cljam/pull/144

Thank you.

totakke commented:
Thank you! The additional commits also seem good, so I've just merged.

alumi commented:
@totakke @federkasten Thank you for your helpful comments! 😹
